sidekick/views.py

```diff
@@ -276,3 +277,55 @@ def post(self, request, pk):
 
         task = archive_turn_conversation.delay(pk, address, reason)
         return Response({"task_id": task.id}, status=status.HTTP_201_CREATED)
+
+
+class ListContactsView(GenericAPIView):
+    """
+    Accepts Org id and query
+    Forwards the query to RapidPro to filter contacts
+    Returns a JsonResponse listing just the uuids of matching contacts
+    """
+
+    queryset = Organization.objects.all()
+
+    def get(self, request, pk, *args, **kwargs):
+        org = self.get_object()
+
+        if not org.users.filter(id=request.user.id).exists():
+            return JsonResponse(
+                data={
+                    "error": "Authenticated user does not belong to specified Organization"
+                },
+                status=status.HTTP_401_UNAUTHORIZED,
+            )
+
+        filter_kwargs = {}
+        field = request.GET.get("field", None)
+        if field:
+            try:
+                value = request.GET["value"]
+            except KeyError:
+                return JsonResponse(
+                    {"error": "'field' param must have corresponding 'value' param"},
+                    status=status.HTTP_400_BAD_REQUEST,
+                )
+            filter_kwargs = {field: value}
+
+        client = TembaClient(org.url, org.token)
+        try:
+            contact_batches = client.get_contacts(**filter_kwargs).iterfetches()
+        except TypeError:
+            return JsonResponse(
+                {
+                    "error": "Invalid 'field' value. Please see RapidPro "
+                    "documentation for valid filters."
+                },
+                status=status.HTTP_400_BAD_REQUEST,
+            )
```

Comment:
Should we also catch any request errors here and forward them on, or do we want those to pop up in Sentry?

Reply:
That is probably a good idea

Reply:
@rudigiesler now that I'm looking at this I'm not sure when this would raise a request error.
Authentication is handled before this point and I think we'd probably want that to raise the error anyway. and invalid fields are handled earlier in the view.
do you have any suggestions?

Reply:
Well, what I meant here is, what if the RapidPro API is down / not responding in time (that's actually another thing, do we want a timeout?), then do we want this raised in sidekick sentry, and a 500 error passed on to the client; or do we want to handle it, and send a different response to the client, and have the client deal with the error?

It should be a `requests.exceptions.RequestException` that gets raised for any errors: https://2.python-requests.org/en/master/user/quickstart/#errors-and-exceptions

Reply:
@rudigiesler the Temba client throws it's own exceptions for these things https://rapidpro-python.readthedocs.io/en/latest/#error-handling
I think some of these exceptions should show up in Sentry since they aren't something the client would be able to fix. So I decided to only forward the rate limit and connection errors, things that the client could probably retry. The Bad Request error I haven't caught because I think the majority of those cases will be covered by the previous catches.